src/main/java/com/daigo/springboot_handson_4/service/MainService.java

```diff
@@ -0,0 +1,158 @@
+package com.daigo.springboot_handson_4.service;
+
+import com.daigo.springboot_handson_4.config.GeoCoderConfig;
+import com.daigo.springboot_handson_4.config.LocalSearchConfig;
+import com.daigo.springboot_handson_4.config.YahooApiConfig;
+import com.daigo.springboot_handson_4.domains.CafeSearchMessenger;
+import com.daigo.springboot_handson_4.domains.geocoder.ContentsGeoCoder;
+import com.daigo.springboot_handson_4.domains.localsearch.LocalSearch;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * SpringbootHandson4Applicationのサービスクラス
+ */
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class MainService {
+    // inject Config class
+    private final YahooApiConfig yahooApiConfig;
+    private final GeoCoderConfig geoCoderConfig;
+    private final LocalSearchConfig localSearchConfig;
+
+    // variables
+    static final String OUTPUT = "json";
+
+    // instances
+    private static final RestTemplate restTemplate = new RestTemplate();
+
+    /**
+     * userLocationから近い位置にあるcafeを探し、その情報を返却するメソッド
+     *
+     * @param userLocation 検索の中心となる地点の名称
+     * @return cafeSearchMessenger 情報を格納したCafeSearchMessenger型のインスタンス
+     */
+    public CafeSearchMessenger searchCafe(String userLocation) {
```

Comment:
```suggestion
    public CafeSearchMessenger searchCafe(final String userLocation) {
```
実は引数にも`final`を指定することができ、
こうすることで「受け付けた引数は関数内で勝手に書き換えられることはない」
と保証＆宣言することができます！